### PR TITLE
feat: bring back FOXy approve

### LINKED ISSUE #4238: Issue when withdrawing Foxy
### Overview

A user is experiencing an issue withdrawing foxy. I am unable to recreate on my wallet, but was able to recreate using users wallet address. User wallet address: 0xCfaba3994C45aa03e479000300801AE35B39c388. After selecting an amount to withdraw and hitting Continue, it gives an error message and wont progress to next screen. 

### References and additional details

1) Log into app.shapeshift and connect to MM
2) Navigate to Fox Opportunities page and note the balance of Fox Staking Token as being zero
3) Select the Manage button for Fox Staking Token and notice there is a balance that has been deposited
4) Select Withdraw and enter amount 
5) Select Continue - notice "Something went wrong" error message appears and the continue button spins continuously. 
image
image
image

### Acceptance Criteria

User should be able to withdraw foxy

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

_No response_

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

tl;dr we assumed that the approval step was not needed for FOXy and removed the components, but it *may* be needed in some specific cases, to send back the reward FOXy back to the staking contract to be able to withdraw the full balance, see:

https://github.com/shapeshift/yieldies/blob/b61e2ca12425488fc437bd1931d8ed6fbfa1bf16/src/contracts/Staking.sol#L489

Reverts the `<Approve />` and related hunks of b1b226cc27e143c9c87a7c2d1d5a6bec358ad114 to bring the approval flow back.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #4238

## Testing

- Test with Frame watch-only user account in #4238
- Ensure you see an approval step for FOX *with this specific account* (none of my own actually needed an approval here)
- Unfortunately, unless you need an approval on one of your own accounts for which you're a signer, it is impossible to test the full flow up to the confirmation... unless you have the private key for that account

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### Prod

image
image

### This diff

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4243** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4243" target="_blank">Graphite</a>  👈
    * **PR #4244** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4244" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4243?utm_source=stack-comment).

File: src/features/defi/providers/foxy/components/FoxyManager/Withdraw/components/Withdraw.tsx [hunk 1]
```diff
@@ -146,17 +170,41 @@ export const Withdraw: React.FC<
...
+        // Check is approval is required for user address
+        const _allowance = await foxyApi.allowance({
+          tokenContractAddress: rewardId,
+          contractAddress,
+          userAddress: accountAddress,
         })
+
+        const allowance = bnOrZero(bn(_allowance).div(bn(10).pow(asset.precision)))
+
+        // Skip approval step if user allowance is greater than or equal requested deposit amount
```
Comment (gomesalexandre) [inline src/features/defi/providers/foxy/components/FoxyManager/Withdraw/components/Withdraw.tsx:182]:
```suggestion
        // Skip approval step if user allowance is greater than or equal requested withdraw amount
```

**Comment by lmyslinski** [comment]
Did a full stake & unstake flow, confirm pops up as expected, all tx also went through
Zrzut ekranu 2023-04-12 o 15 24 18
image
